### PR TITLE
python3 compatibility

### PR DESCRIPTION
**Pull request by geosciz**
Hi @rabernat,

I notice that `setup.py` has an issue of `ValueError: can't have unbuffered text I/O` if you install the `floater` package in a python3 environment. I thus google this problem, find a solution in this [link](https://code.djangoproject.com/ticket/20815). Then I make some modifications and submit this pull request to add more compatibility for python3 installation.

Best,

@geosciz

### CODE REVIEW

**Comment by rabernat** [comment]
Fantastic! Since you are fixing the python 3 compatibility problems, could you try re-enabling the python3 tests in .travis.yml? That will show us if there are any other python3 issues:
https://github.com/rabernat/floater/blob/master/.travis.yml#L13

File: floater/test/test_hexarray.py [hunk 1]
```diff
@@ -16,14 +17,24 @@ def setUp(self):
...
+                        self.assertEqual(len(ha.neighbors(n)), 0)
+            if PY3:
+                for n in range(9):
+                    if n == 4:
+                        self.assertSequenceEqual(
+                            list(ha.neighbors(n)),
+                            [0, 1, 5, 7, 6, 3]
+                        ) 
+                    else:
+                        self.assertEqual(len(ha.neighbors(n)), 0)
```
Comment (rabernat) [inline floater/test/test_hexarray.py]: You definitely don't want to have different code paths for python 2 vs 3

File: floater/test/test_hexarray.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import unittest
 import numpy as np
 from floater import hexgrid
+from six import PY2, PY3
```
Comment (rabernat) [inline floater/test/test_hexarray.py]: don't do this

File: floater/test/test_hexarray.py [hunk 3]
```diff
@@ -16,14 +17,24 @@ def setUp(self):
...
-            for n in xrange(9):
-                if n == 4:
-                    self.assertSequenceEqual(
-                        list(ha.neighbors(n)),
-                        [0, 1, 5, 7, 6, 3]
-                    )
-                else:
-                    self.assertEqual(len(ha.neighbors(n)), 0)
+            if PY2:
+                for n in xrange(9):
```
Comment (rabernat) [inline floater/test/test_hexarray.py]: just use `range` for both 2 and 3

File: setup.py [hunk 4]
```diff
@@ -40,7 +40,7 @@ def check_for_openmp():
         cc = os.environ['CC']
     except KeyError:
         cc = 'gcc'
-    with open(filename, 'w', 0) as file:
+    with open(filename, 'w', 1) as file:
```
Comment (rabernat) [inline setup.py]: what does the `1` vs `0` mean here?

File: floater/test/test_hexarray.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import unittest
```
Comment (rabernat) [inline floater/test/test_hexarray.py]: you want to add
```python
from __future__ import print_function
```
at the very top of all files

File: setup.py (hunk 4 above)
Comment (geosciz) [inline setup.py]: this is for the optional argument `buffering` in the function `open`. `0` means unbuffered which causes problems in Python 3, and `1` means line buffered. A negative value here say `-1` means system default settings.

Comment (rabernat) [inline setup.py]: thanks for the clarification!

**Comment by rabernat** [comment]
This is excellent! Thanks so much Ci!

**Pull request merged**